src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -897,6 +905,27 @@ type DiagnosticItem =
                 "This expression may be short-circuited, and operation calls may not be executed."
             | WarningCode.DeprecationWithRedirect -> "{0} has been deprecated. Please use {1} instead."
             | WarningCode.DeprecationWithoutRedirect -> "{0} has been deprecated."
+            | WarningCode.UnsupportedResultComparison ->
```

Comment:
What are the 4 additional arguments to this message?

Reply:
Reading more, I believe the arguments are callable name, source file, start line, start column. The format with [source : line : column] seems a bit cryptic to me. Would it make sense to instead just make it another sentence that also includes the callable name? That would be more in line with how other diagnostics look I think.

Reply:
I don't think there are any other diagnostics that refer to a different file. However, it looks like the format for file locations displayed by the compiler is `file(line,column)`.

Reply:
Oh, it seems like `file(line,column)` is only from `dotnet build`. The compiler itself shows

> File: filename.qs
> Position: [ln X, cn Y]

Reply:
The compiler itself has two modes to display diagnostic: one that is intended for best human readability, and then one mode that formats the diagnostics in a way that is picked up by the IDE (that only happens if the emitted diagnostics follow a specific format). 
Whatever format you choose is fine by me; that's just my two cents.

Reply:
I can't really think of a good way to refer to another file and location from inside a diagnostic message. Ideally, we would just set the diagnostic location itself instead of inserting it into the message, but the compiler doesn't support that at the moment. I think copying the compiler's human-readable output format is OK for now.